ibaby_smarthome_multinode/README.md

```diff
@@ -0,0 +1,306 @@
+
+|  folder/file        |            Function                                           |
+| ------------------- | ------------------------------------------------------------- |
+|  driver             |        source code of drivers                                 |
+|  function           |        source code of function modules                        |
+|  lwm2m_client       |        source code of LwM2M Client                            |
+|  FreeRTOSConfig.h   |        header file of FreeRTOS configurations                 |
+|  main.c             |        main entry of embARC Application                       |
+|  makefile           |        Makefile of embARC Application                         |
+
+#### Makefile
+
+- Selected FreeRTOS here, then you can use [FreeRTOS API][39] in your application:
+
+		# Selected OS
+		OS_SEL ?= freertos
+
+- Target options about EMSK and toolchain:
+
+		BOARD ?= emsk
+		BD_VER ?= 22
+		CUR_CORE ?= arcem11d
+		TOOLCHAIN ?= gnu
+
+- Reset the heap and stack size for LwM2M, make sure they are big enough for your application:
+
+		##
+		# HEAP & STACK SETTINGS
+		# For LwM2M Stack Usage
+		##
+		HEAPSZ ?= 81920
+		STACKSZ ?= 81920
+
+- The relative series of the root directory, here the path of the Makefile is `./embarc_osp/application/ibaby_smarthome_multinode/src/lamp_node/makefile`:
+
+		#
+		# root dir of embARC
+		#
+		EMBARC_ROOT = ../../../..
+
+- The middleware used in your application:
+
+		MID_SEL = common lwip-contrib wakaama fatfs lwip
+
+	common for baremetal function, lwip-contrib and wakaama for LwM2M, fatfs for file system, lwip for wifi.
+
+	You might be wondering about **how wifi works?** There is nothing about it in the lamp node or wearable node application. Goto `./embarc_osp/board/board.c`, and you'll solve the problem:
+
+		#if defined(OS_FREERTOS) && defined(MID_LWIP)
+		static void task_wifi(void *par)
+		{
+		...
+		}
+
+	Wifi works as a independent task on the FreeRTOS, so you ought to select *freertos* for OS_SEL and include *lwip* in the MID_SEL. Then, the task for wifi will start to work automatically.
+
+	![wifi_connected_info][4]
+
+- Directoties of source files and header files, notice that it **can not recursive**:
+
+		# application source dirs
+		APPL_CSRC_DIR = . ./lwm2m_client ./driver/acceleration ./driver/body_temperature ./driver/heartrate ./driver/timer ./function/ ./function/lwm2m ./function/print_msg ./function/process_acc ./function/process_hrate
+		APPL_ASMSRC_DIR = .
+
+		# application include dirs
+		APPL_INC_DIR = . ./lwm2m_client ./driver/acceleration ./driver/body_temperature ./driver/heartrate ./driver/timer ./function/ ./function/lwm2m ./function/print_msg ./function/process_acc ./function/process_hrate
+
+See `./embarc_osp/doc/embARC_Document/html/page_example.html`, **"Options to Hard-Code in the Application Makefile"** for more detailed information about **Makefile Options**.
```

Comment:
./embarc_osp/doc/embARC_Document/html/page_example.html ->  http://embarc.org/embarc_osp/doc/embARC_Document/html/page_example.html 

You can use online links